"complete" event being fired before derived event (task) being completed

In v3.1.0, the "completed" event is being fired after the first handler is completed, but the handler also fires a task event (which hasn't finished when the complete is fired). According to [the doc](http://circuits.readthedocs.org/en/latest/api/circuits.core.events.html#circuits.core.events.Event), I would expect the "completed" event only to be fired after the task and derived fired events have been handled.


Example output:
```
<NewInputPath[*] ('data.db' )>
<NewInputPath[*] ('files_sent.log' )>
<task[<Worker/Compressor 10093:MainThread (queued=0) [S]>] (<bound method _CompressorJob._do_task of <_CompressorJob/* 10093:MainThread (queued=0) [S]>>, <fcb.processing.models.FileInfo.FileInfo instance at 0x7f2d5c0a8d88> )>
<NewInputPath_complete[*] (<NewInputPath[*] ('data.db' )>, None )>
<task[<Worker/Compressor 10093:MainThread (queued=0) [S]>] (<bound method _CompressorJob._do_task of <_CompressorJob/* 10093:MainThread (queued=0) [S]>>, <fcb.processing.models.FileInfo.FileInfo instance at 0x7f2d5c0a8d40> )>
<NewInputPath_complete[*] (<NewInputPath[*] ('files_sent.log' )>, None )>
```

@keli 
The tests are currently failing:
https://travis-ci.org/circuits/circuits/jobs/142853110
```
    def _fire(self, event, channel, priority=0):
        # check if event is fired while handling an event
        th = (self._executing_thread or self._flushing_thread)
>       if thread.get_ident() == (th.ident if th else None) and \
                not isinstance(event, signal):
E               AttributeError: 'int' object has no attribute 'ident'
```

--- a/circuits/core/manager.py
+++ b/circuits/core/manager.py
@@ -574,7 +574,7 @@ class Manager(object):
         # events. Note that _flush can be called recursively.
         old_flushing = self._flushing_thread
         try:
-            self._flushing_thread = thread.get_ident()
+            self._flushing_thread = current_thread()
             if self._flush_batch == 0:
                 self._flush_batch = len(self._queue)
             while self._flush_batch > 0:

